> ```
>    > --- FAIL: TestGoBuildWithKOCACHE (5.82s)
>    >     gobuild_test.go:1062: Build() = build: go build: exit status 1: # runtime/cgo
>    >         gcc: error: unrecognized command-line option '-m64'
> ```
> 
> `TestGoBuildWithKOCACHE` tries to build for "all arch" from ko. This fails on aarch64 it seems because it's not the "correct" gcc (or something is missing), not sure how to tell nixpkgs to add it during the tests. (see [confluentinc/confluent-kafka-go#898 (comment)](https://github.com/confluentinc/confluent-kafka-go/issues/898#issuecomment-1704003644) or https://forum.golangbridge.org/t/cross-compiling-go/34372 maybe ?)

Does setting `env.CGO_ENABLED = 0;` help?

--- a/pkgs/by-name/ko/ko/package.nix
+++ b/pkgs/by-name/ko/ko/package.nix
@@ -9,21 +9,23 @@
 
 buildGo123Module rec {
   pname = "ko";
-  version = "0.15.4";
+  version = "0.17.1";
 
   src = fetchFromGitHub {
     owner = "ko-build";
     repo = pname;
     tag = "v${version}";
-    hash = "sha256-MeFoy2WoPsJIgUhpzt/4sEP6J9lM4nsSAK2VZiTS7jo=";
+    hash = "sha256-OQtYyokARrjaf0MWQ0sMqJPb+C5pRkKFumAmtxS4SBo=";
   };
 
-  vendorHash = "sha256-n/NbbitSyjl05gESPVG3Uv2ek1U0Cd2fQqcxBhDKULU=";
+  vendorHash = "sha256-YQggwX6fUsfZMM+GdgeNAIHkfX84FMF84xHsP/SNiS4=";
 
   nativeBuildInputs = [ installShellFiles ];
 
   # Pin so that we don't build the several other development tools
   subPackages = ".";
+
+  env.CGO_ENABLED = 0;
 
   ldflags = [
     "-s"
@@ -32,8 +34,8 @@ buildGo123Module rec {
   ];
 
   checkFlags = [
-    # requires docker daemon
-    "-skip=TestNewPublisherCanPublish"
+    # requires docker daemon, pulls and builds delve debugger
+    "-skip=(TestNewPublisherCanPublish|TestDebugger)"
   ];
 
   nativeCheckInputs = [ gitMinimal ];
@@ -64,13 +66,13 @@ buildGo123Module rec {
   meta = with lib; {
     homepage = "https://github.com/ko-build/ko";
     changelog = "https://github.com/ko-build/ko/releases/tag/v${version}";
-    description = "Build and deploy Go applications on Kubernetes";
+    description = "Build and deploy Go applications";
     mainProgram = "ko";
     longDescription = ''
       ko is a simple, fast container image builder for Go applications.
       It's ideal for use cases where your image contains a single Go application without any/many dependencies on the OS base image (e.g. no cgo, no OS package dependencies).
       ko builds images by effectively executing go build on your local machine, and as such doesn't require docker to be installed. This can make it a good fit for lightweight CI/CD use cases.
-      ko also includes support for simple YAML templating which makes it a powerful tool for Kubernetes applications.
+      ko makes multi-platform builds easy, produces SBOMs by default, and includes support for simple YAML templating which makes it a powerful tool for Kubernetes applications.
     '';
     license = licenses.asl20;
     maintainers = with maintainers; [

